src/main/java/com/google/sps/data/OrganizationUpdater.java

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Arrays;
+import com.google.appengine.api.datastore.Entity;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+
+public final class OrganizationUpdater {
+
+  private Entity entity;
+
+  public OrganizationUpdater(Entity entity) {
+     this.entity = entity;
+  }
+
+  public Entity getEntity() {
+      return this.entity;
+  }
+
+  public void updateOrganization(HttpServletRequest request, boolean isMaintainer, boolean isModerator) throws IllegalArgumentException{
```

Comment:
What do you think of accepting a GiverUser instead of these booleans?

Reply:
So right now GiverUser does not include an IsModerator check or field and its still a work in progress so it would be hard to test. Should I go ahead and do it anyways or add a TODO?

Reply:
There is currently a GiverUser checked in right; what are your concerns here? Is there a middle ground option?

Reply:
To be clear: the heart of my question is basically: if we have this abstraction for "User," why aren't we using it and extending it? I want to make sure the work we do complements each other; that we're not just pulling in different directions.

Reply:
My concern is that at the moment there GivrUser does not have a way of fulfilling the isModerator boolean check. On the edit org servlet side it would pass in request, GivrUser.isMaintainer, true since the edit functionality would only be displayed to users who are at least moderators in the first place. But when it comes from the Add Organization side it does not care who the user is since the registration form only takes care of certain params which is why it calls it with request, false, false. If I replaced this with a GiverUser then isMaintainer and isModerator would both have to result in false in order for the isApproved and ModeratorList params to not be checked which could be false since they could be a maintainer and they would technically be the first moderator for the org that they are submitting the request for. The way im looking at it now these checks are more for specifying which servlet is calling this function. I am thinking that if we did pass in a GiverUser param then we would also have to pass in if its calling update for registering or not so it ignores the user which seems redundant

Reply:
It seems like an error prone pattern to ask the caller of this function to sometimes lie about the authorization level of the user! Is there another way we could solve this?

Reply:
Consider: right now you are planning on sending down just "true" for isModerator right?

If we assume GIvrUser will do this can you can store this as local variables. For now you can always set isModerator to false.

On the consideration of different behavior for the caller: you should be careful about using  isModerator/isMaintainer to signify who the caller is. If you really have to change behavior based on who is calling the function, then consider a different parameter. If you did want to continue to use the call as if the caller is always unauthenticated, then you can create a subclass of the GivrUser or somehow consruct a GivrUser that guarantees it has no permissions.